go-controller/pkg/factory/factory.go

```diff
@@ -406,6 +394,14 @@ func NewOVNKubeControllerWatchFactory(ovnClientset *util.OVNKubeControllerClient
 		}
 	}
 
+	if util.IsNetworkSegmentationSupportEnabled() {
```

Comment:
because master watch factory calls 
wf, err := NewOVNKubeControllerWatchFactory(ovnClientset.GetOVNKubeControllerClientset())

and we create it there now